staging/README.md

```diff
@@ -0,0 +1,89 @@
+# Introduction
+
+Welcome to the staging tree!
+
+This tree contains experimental code that does not (yet) meet the quality
+requirements of RIOT.
+
+The idea of this tree is to provide a place to code that offers functionality,
+but has not (yet) received the polishing and refinement that is usually
+required in RIOT. staging/ is also a place where new ideas can be shared early
+with a wider audience in order to facilitate easier collaboration.
+
+**WARNING**
+
+Do not depend on anything within this subfolder!  Everything in here might be
+broken or break or be changed or removed at any time, and everything is
+officially unsupported (but may be supported by one or more individuals)! API's
+from this tree might change without notice.
+
+## How to use
+
+Please re-read the warning above.
+If you still feel brave enough to use code contained in here, add
+
+    ENABLE_STAGING=1
+
+to your application's Makefile.
+
+## Terminology
+
+"main tree" is used to describe the regular RIOT sources (everything outside
+staging/).
+
+"broken" means, it doesn't compile or work anymore.
+
+## Rules
+
+- no dependencies from main tree into staging are allowed
+- commits MUST NOT change files both in and outside of staging/
+- if anything in staging/ breaks due to a change in master, it will be marked
+  as broken
+- the original contributor(s) is/are considered maintainer(s) for a module
+- non-trivial changes SHOULD be ACK'ed by a module's maintainer(s)
+- PR's to staging/ are labeled with "staging"
+- PR's changing both staging and main tree code are *not* labeled "staging"
+- each staging module has a README.md as described below
+
+## Procedures
+
+### Reviewing a change to staging/
+
+Make sure above rules are followed. For new contributions, only accept code
+that has at least a remote chance of getting merged into the main tree at some
+point, e.g., filter out nonsense contributions. Unless the contribution
+consists of a header-only API that's in development, check that all code gets
+compiled by an accompanying test application.
+
+### Mark as "broken"
+
+Modules can be marked as broken in staging/staging.mk.
+Add a comment with a date and a reason, e.g.,
+
+    BROKEN += heart   # 2018/12/24: broken through change in friend API (#12345)
+
+If a contribution to the main tree fails to build because of a module in
+staging/, it is perfectly valid to mark that module as "broken".
```

Comment:
Not sure I get this. Can you give a rough example where a PR to the maint tree would be affected?
What is the consequence of marking a module in staging as broken?

Reply:
Hm, the idea here is to define how to deal with the mainanance burden of suboptimally integrated code.
So if for example a build system improvement breaks a driver in staging/, that should not block the improvement PR. So to get the PR to compile it is alright to disable a staging/ module.

The consequence is that the staging module (and its dependees) won't be built anymore, and if noone cares enough to fix it, it'll be removed after N release cycles.

I hope that this will not be used in trivial cases (e.g., a slight API change that modules can easily be adapted to), but where half-integrated code prevents the main tree from advancing.

Reply:
I tried to put this into the file, ok like this?

Reply:
Clearer. But I suppose that marking it as broken means that the CI will not compile it anymore, or did I get something wrong?
If so, then it would be good to indicate that explicitly above when we introduce the "broken" list ?

Reply:
did.